File: src/Benchmarking.java
import java.util.Random;



public class Benchmarking {
    private MetodosOrdenamiento MetodosOrdenamiento;
    public Benchmarking(){
        long currentMillis = System.currentTimeMillis();//sacar la fecha
        long currentNano = System.nanoTime();
        System.out.println(""+currentMillis);
        System.out.println("" +currentNano);

        MetodosOrdenamiento =new MetodosOrdenamiento();

        int [] arreglo= generarArregloAleatorio(1000);
        Runnable tarea = ()-> MetodosOrdenamiento.burbujaTradicional(arreglo);
        double tiempoDuracionMilis= medirConCurrerntTimeMiles(tarea);
        double tiempoDuracionNano= medirNanoTieme(tarea);

        System.out.println("tiempo en milisegundos: "+ tiempoDuracionMilis);
        System.out.println("tiempo en nanosegundos: "+ tiempoDuracionNano);

    }


    private int[] generarArregloAleatorio(int tamaño) {
        int []array =new int[tamaño];
        Random random = new Random();
        for (int i=0;i<tamaño;i++) {
            array[i]= random.nextInt(100000);
        }
        return array; 
    }

    public double medirConCurrerntTimeMiles(Runnable tarea){
        long inicio = System.currentTimeMillis();
        tarea.run();
        long fin = System.currentTimeMillis();
        double tiempoSegundos = ((fin-inicio) / 1000.0);
        return tiempoSegundos;
    }

    public double medirNanoTieme(Runnable tarea){
        long inicio = System.nanoTime();
        tarea.run();
        long fin= System.nanoTime();
        double tiempoSegundos= ((fin-inicio) / 1000000000.0);
        return tiempoSegundos;
    }
    
}

File: src_py/benchmarking.py
from metodos_ordenamiento import MetodosOrdenamiento
#import metodos_ordenamiento as mO
import random
import time
#declaracion de la clase
class benchmarking:
    
    def __init__(self): #metodo constructor
        """
        print("Benchmarking Instanciado")
        self.mO =MetodosOrdenamiento()

        arreglo=self.buil_arreglo(10000)

        tarea= lambda:self.mO.sort_bubble(arreglo)

        tiempoM=self.contar_con_current_time_milles(tarea)
        tiempoN=self.contar_con_nano_times(tarea)
        #print(f"tiempo en milisegundos: {tiempoM}")
        print(f"tiempo en nanosegundos: {tiempoN}")

        tarea1= lambda:self.mO.sort_burbuja_mejorado_optimizado(arreglo)
        tiempoM=self.contar_con_current_time_milles(tarea1)
        tiempoN=self.contar_con_nano_times(tarea1)
        #print(f"tiempo en milisegundos: {tiempoM}")
        print(f"tiempo en nanosegundos: {tiempoN}")
        
        tarea2= lambda:self.mO.sort_seleccion(arreglo)
        tiempoM=self.contar_con_current_time_milles(tarea2)
        tiempoN=self.contar_con_nano_times(tarea2)
        #print(f"tiempo en milisegundos: {tiempoM}")
        print(f"tiempo en nanosegundos: {tiempoN}")
        
        """
        
    def medir_tiempo(self, funcion, array):
        inicio= time.perf_counter()
        funcion(array)
        fin= time.perf_counter()
        return fin -inicio
    
    def buil_arreglo(self, tamaño):
        arreglo=[]
        for i in range(tamaño):
            numero= random.randint(0,99999)
            arreglo.append(numero)
        return arreglo

    #importar archivo tipo time
    #milisegundos en segundos con #x= time.time()
    #nanosegindos con x= time.time_ns()
    def contar_con_current_time_milles(self,tarea):
        inicio = time.time()
        tarea()
        fin = time.time()
        tiempoSegundos = fin-inicio
        return tiempoSegundos

    def contar_con_nano_times(self,tarea):
        inicio =time.time_ns()
        tarea()
        fin=time.time_ns()
        tiempons=(fin -inicio)/1_000_000_000.0
        return tiempons

